File: AiStudio4/Core/Models/SystemPrompt.cs



namespace AiStudio4.Core.Models
{
    public class SystemPrompt
    {
        public string Guid { get; set; } = System.Guid.NewGuid().ToString();
        public string Title { get; set; } = string.Empty;
        public string Content { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public bool IsDefault { get; set; } = false;
        public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
        public DateTime ModifiedDate { get; set; } = DateTime.UtcNow;
        public List<string> Tags { get; set; } = new List<string>();
        public List<string> AssociatedTools { get; set; } = new List<string>(); // Tool GUIDs
        public string AssociatedUserPromptId { get; set; } = string.Empty; // Associated User Prompt GUID
        public List<string> AssociatedMcpServers { get; set; } = new List<string>(); // MCP Server IDs

        // NEW: Model associations
        public string PrimaryModelGuid { get; set; } = string.Empty; // Associated primary model GUID
        public string SecondaryModelGuid { get; set; } = string.Empty; // Associated secondary model GUID
        public bool IncludeGitDiff { get; set; } = false; // Whether to include git diff when this prompt is applied
        
        // NEW: Project association
        public string ProjectGuid { get; set; } = string.Empty; // Associated project GUID
    }
}

File: AiStudio4/AiStudioClient/src/types/systemPrompt.ts
export interface SystemPrompt {
  guid: string;
  title: string;
  content: string;
  description: string;
  isDefault: boolean;
  createdDate: string;
  modifiedDate: string;
  tags: string[];
  associatedTools: string[]; // Tool GUIDs
  associatedUserPromptId?: string; // Associated User Prompt GUID
  associatedMcpServers?: string[]; // MCP Server IDs
  primaryModelGuid?: string; // Associated primary model GUID
  secondaryModelGuid?: string; // Associated secondary model GUID
  includeGitDiff?: boolean; // Whether to include git diff when this prompt is applied
}

export interface SystemPromptFormValues {
  title: string;
  content: string;
  description: string;
  tags: string[];
  isDefault?: boolean;
  associatedTools: string[]; // Tool GUIDs
  associatedUserPromptId?: string; // Associated User Prompt GUID
  associatedMcpServers?: string[]; // MCP Server IDs
  primaryModelGuid?: string; // Associated primary model GUID
  secondaryModelGuid?: string; // Associated secondary model GUID
  includeGitDiff?: boolean; // Whether to include git diff when this prompt is applied
}